bindings/python/test/test_converters.py

```diff
@@ -25,24 +28,167 @@
 
 
 def test_coerce_to_datetime_success_instant():
-    value = Instant.date_time(DateTime(2020, 1, 1), Scale.UTC)
-    assert coerce_to_datetime(value) == datetime(2020, 1, 1, tzinfo=timezone.utc)
+    value = Instant.date_time(DateTime(2020, 1, 2, 3, 4, 5, 123, 456), Scale.UTC)
+    assert coerce_to_datetime(value) == datetime(
+        2020, 1, 2, 3, 4, 5, 123456, tzinfo=timezone.utc
+    )
+
+    value = Instant.date_time(DateTime(2020, 1, 2, 3, 4, 5, 123, 456), Scale.TAI)
+    assert coerce_to_datetime(value) == datetime(
+        2020, 1, 2, 3, 3, 28, 123456, tzinfo=timezone.utc
+    )
 
 
 def test_coerce_to_datetime_success_datetime():
-    value = datetime(2020, 1, 1, tzinfo=timezone.utc)
+    value = datetime(2020, 1, 2, 3, 4, 5, 123456, tzinfo=timezone.utc)
     assert coerce_to_datetime(value) == value
 
+    value = datetime(
+        2020, 1, 2, 3, 4, 5, 123456, tzinfo=timezone(timedelta(seconds=3600))
+    )
+    assert coerce_to_datetime(value) == value
+
+
+def test_coerce_to_datetime_success_iso():
```

Comment:
Maybe add a test for coercing "2020-01-02T03:04:05.123456Z"
And also without milliseconds/microseconds?

Reply:
Added Z format with and without decimals for all 3 functions.